test/modules/WebSocket/exchange/websocket.server.connect.test.ts

```diff
@@ -118,3 +118,35 @@ it('throw an error when connecting to a non-existing server', async () => {
     expect(errorListener).toHaveBeenCalledTimes(1)
   })
 })
+
+it('inherits the "binaryType" from the mock WebSocket', async () => {
+  const clientMessageListener = vi.fn<[ArrayBuffer]>()
+  const interceptorMessageListener = vi.fn<[ArrayBuffer]>()
+
+  wsServer.on('connection', (ws) => {
+    // Set the "binaryType" for the "ws" package also
+    // so it sends ArrayBuffer and not internal "nodebuffer".
+    ws.binaryType = 'arraybuffer'
+    ws.send(new TextEncoder().encode('hello'))
+  })
+
+  interceptor.once('connection', ({ server }) => {
+    server.connect()
+    server.addEventListener('message', (event) => {
+      interceptorMessageListener(event.data)
+    })
+  })
+
+  const ws = new WebSocket(getWsUrl(wsServer))
+  // Set a custom binary type for this socket instance.
+  ws.binaryType = 'arraybuffer'
```

Comment:
We have to ensure the compatibility between the two WebSocket instances. But the `binaryType` is the only property that the consumer can override. Everything else, like `bufferedAmount` or `extensions` is read-only.